gnomad_qc/v4/annotations/generate_variant_qc_annotations.py

```diff
@@ -316,14 +336,152 @@ def run_generate_sib_stats(
+            for tp, n in tp_map.items()
+            for group in GROUPS
+        },
+        fail_hard_filters=(ht.QD < 2) | (ht.FS > 60) | (ht.MQ < 30),
+        singleton=ht.AC_release_raw == 1,
+        ac_raw=ht.AC_high_quality_raw,
+        ac=ht.AC_release,
+        ac_unrelated_raw=ht.AC_unrelated_raw,
+    )
+
+    if impute_features:
+        ht = median_impute_features(ht, {"variant_type": ht.variant_type})
+
+    ht = ht.repartition(n_partitions, shuffle=False)
+    ht = ht.checkpoint(
+        hl.utils.new_temp_file("variant_qc_annotations", "ht"), overwrite=True
+    )
+
+    summary = ht.group_by(
+        *TRUTH_DATA, *[f"{tp}_{group}" for tp in tp_map for group in GROUPS]
+    ).aggregate(n=hl.agg.count())
+    logger.info("Summary of truth data annotations:")
+    summary.show(-1)
+
+    return ht
+
+
+def get_tp_ht_for_vcf_export(
+    ht: hl.Table,
+    transmitted_singletons: bool = False,
+    sibling_singletons: bool = False,
+) -> Dict[str, hl.Table]:
+    """
+    Get Tables with raw and adj true positive variants to export as a VCF for use in VQSR.
+
+    :param ht: Input Table with transmitted singleton and sibling singleton information.
+    :param transmitted_singletons: Whether to include transmitted singletons in the
+        true positive variants.
+    :param sibling_singletons: Whether to include sibling singletons in the true
+        positive variants.
+    :return: Dictionary of 'raw' and 'adj' true positive variant Tables.
+    """
+    if not transmitted_singletons and not sibling_singletons:
+        raise ValueError(
+            "At least one of transmitted_singletons or sibling_singletons must be set "
+            "to True"
+        )
+    tp_hts = {}
+    for group in GROUPS:
+        filter_expr = False
+        if transmitted_singletons:
+            filter_expr = ht[f"transmitted_singleton_{group}"]
+        if sibling_singletons:
+            filter_expr = filter_expr | ht[f"sibling_singleton_{group}"]
+
+        filtered_ht = ht.filter(filter_expr).select().select_globals()
+        filtered_ht = filtered_ht.checkpoint(
+            hl.utils.new_temp_file("true_positive_variants", "ht"),
+            overwrite=True,
+        )
+        logger.info(
+            "True positive %s Table for VCF export contains %d variants",
+            group,
+            filtered_ht.count(),
+        )
+        tp_hts[group] = filtered_ht
+
+    return tp_hts
+
+
 def get_variant_qc_annotation_resources(
-    test: bool, overwrite: bool
+    test: bool, overwrite: bool, true_positive_type: Optional[str] = None
 ) -> PipelineResourceCollection:
     """
     Get PipelineResourceCollection for all resources needed in the variant QC annotation pipeline.
 
     :param test: Whether to gather all resources for testing.
     :param overwrite: Whether to overwrite resources if they exist.
+    :param true_positive_type: Type of true positive variants to use for true positive
+        VCF path resource. Default is None.
```

Comment:
```suggestion
        VCF path resource when exporting. Default is None.
```

A bit unclear abt code - does it not export the True Positive VCFs if this isn't passed? If so, may want to mention in params. If I'm just misreading it though, then nvm

Reply:
`get_variant_qc_annotation_resources` is only getting the resources and not actually performing the export

Reply:
ah sorry, makes sense now